### PR TITLE
Update testing strategy

### PR DESCRIPTION
**Pull request by kroenlein**
This PR has no changes to library functionality.

In this PR, we've 
* updated some library dependencies to better support Python 3.11
* modified the install pattern to automatically test against latest Pint and Pandas
* prevented building libraries (in particular Pandas) during testing to keep things speedy

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,17 +7,10 @@ python:
 - '3.10'
 - '3.11'
 env:
-- PINT_VERSION=0.18
-- PINT_VERSION=0.21
-jobs:
-  exclude:
-    - python: 3.7
-      env: PINT_VERSION=0.21
+  - UPGRADES="pint --upgrade pandas --upgrade"
```
Comment (btreecat) [inline .travis.yml:10]: `--upgrade` assumes the package is already installed. I don't think you need to flag those packages for upgrade and the `--upgrade` flag should come after install but before the package and would only be needed once IIRC. 

Example call: https://pip.pypa.io/en/stable/installation/#upgrading-pip

**Review by btreecat — CHANGES_REQUESTED** [review]
I think we can clean up the pip install command, it might need to be broken out into two commands if you want to install a minimum, then force upgrade to a later version.

File: .travis.yml [hunk 2]
```diff
@@ -7,17 +7,12 @@ python:
...
 - '3.11'
 env:
-- PINT_VERSION=0.18
-- PINT_VERSION=0.21
-jobs:
-  exclude:
-    - python: 3.7
-      env: PINT_VERSION=0.21
+  - UPGRADES="-U pint pandas"
+  - UPGRADES=""
```
Comment (btreecat) [inline .travis.yml:11]: Is this reassigning `UPGRADES` to the empty string?

**Review by btreecat — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (kroenlein) [reply .travis.yml:11]: No, this is saying run once with the upgrades, and once using the versions specified in the requirements files.

https://docs.travis-ci.com/user/languages/python/#testing-against-multiple-versions-of-dependencies-eg-django-or-flask

**Review by btreecat — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (btreecat) [reply .travis.yml:11]: TIL! Thanks for the new knowledge :smile:

**Pull request merged**